rotkehlchen/tests/unit/accounting/test_staking.py

```diff
@@ -271,3 +278,143 @@ def test_validator_exit_pnl(
         ],
     )
     assert accountant.pots[0].pnls.taxable == ONE  # 1 ETH
+
+
+@pytest.mark.parametrize('ethereum_accounts', [['0x0fdAe061cAE1Ad4Af83b27A96ba5496ca992139b']])
+@pytest.mark.parametrize('should_mock_price_queries', [True])
+@pytest.mark.parametrize('default_mock_price_value', [FVal(3000)])  # ETH price at $3000
+@pytest.mark.parametrize('db_settings', [{
+    'eth_staking_taxable_after_withdrawal_enabled': True,
+}])
+def test_eth_withdrawal_processing(accountant: Accountant, ethereum_accounts: list[ChecksumEvmAddress]) -> None:  # noqa: E501
+    """Test ETH withdrawal processing with all different scenarios in realistic order:
+    1. Distributing validator withdrawal (profit = amount)
+    2. Accumulating validator skimming withdrawal (profit = full amount)
+    3. Distributing validator exit (profit = amount - MIN_EFFECTIVE_BALANCE)
+    4. Accumulating validator exit (profit = amount - MAX_EFFECTIVE_BALANCE)
+    5. Accumulating validator exit (loss)
+    """
+    v_distributing = 1001  # Distributing validator
+    v_accum = 1002  # Accumulating validator
+    v_accum_2 = 1003  # Accumulating validator
+
+    withdraw_address = ethereum_accounts[0]
+    events = [
+        # 1. Distributing validator withdrawal (profit = amount)
+        EthWithdrawalEvent(
+            validator_index=v_distributing,
+            timestamp=TimestampMS(1689000000000),
+            amount=FVal(2),
+            withdrawal_address=withdraw_address,
+            is_exit=False,
+        ),
+        EvmEvent(
+            tx_hash=make_evm_tx_hash(),
+            sequence_index=1,
+            timestamp=TimestampMS(1689000001000),
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.REMOVE_ASSET,
+            asset=A_ETH,
+            amount=(withdrawal_amount := FVal('0.5')),
+            location_label=withdraw_address,
+            notes=f'Request to withdraw {withdrawal_amount} ETH from validator {v_accum}',
+            counterparty=CPT_ETH2,
+            address=WITHDRAWAL_REQUEST_CONTRACT,
+            extra_data={'validator_index': v_accum},
+        ),
+        # 2. Accumulating validator partial withdrawal (profit = 0)
```

Comment:
Write all comments like this to not waste lines. It's in the whole file. Won't repeat it.

```suggestion
        ), # 2. Accumulating validator partial withdrawal (profit = 0)
```